### PR TITLE
Honor embed query string parameter

### PR DESCRIPTION
**Pull request by RobinIsTheBird**
On map page:
* Add embed context variable.
* Prune header and footer HTML if embed.
* Add an embed class to wrapper.
* Adjust data for Add a Tree or Resource buttons.
* Fix up button anchors
* For embed, don't include the quick edit button.

Style:
* Style adjustments for .wrapper.embed.
* Eliminated ambiguous "map" css class.

JavaScript:
* Save the embed query string parameter in the mapPage.
* If embed is true, fix up the add button's href,
  and let the browser do its thing.
* Change Add to open a new tab or window,
  and preserve zoom.
* Change details links to open a new tab or window.

Add a test to make sure we still get the page when
?embed= is in the query string.

--

Connects to #2736

### CODE REVIEW

**Comment by RobinIsTheBird** [comment]
Note, travis is still failing on flake8 of exporter/tests, nothing to do with this change. Everything else passes.

File: opentreemap/treemap/templates/instance_base.html [hunk 1]
```diff
@@ -158,9 +161,14 @@
            disabled="disabled"><i class="icon-plus"></i> {% trans "Add a" %} {{ term.Resource.singular }}</a>
         {% endif %}
       <a class="btn btn-primary addBtn"
-         data-action='addtree'
+         {% if embed %}
+         target="_blank"
+         {% else %}
          data-feature="add_plot"
```
Comment (jwalgran) [inline opentreemap/treemap/templates/instance_base.html]: Is it important that we remove these data attributes here if we are in embed mode? The "Add Resource" button only has an if embed/endif around the `target="_blank"`. We may want to keep them the same for consistency.

**Comment by jwalgran** [comment]
This is working well. Nice job.

I did find a pair of related edge cases. If the treemap is suspended, the suspension page needs to hide the header and add a `target="_blank"` attribute to the "login" link if `embed=true`

screen shot 2016-09-07 at 11 20 05 am

If the instance is not publicly accessible, the iframe gets redirected to the login page. We should probably redirect to `/not-available` instead if `embed=true`

screen shot 2016-09-07 at 11 17 09 am

File: opentreemap/treemap/templates/instance_base.html (hunk 1 above)
Comment (RobinIsTheBird) [inline opentreemap/treemap/templates/instance_base.html]: `data-action` is still present, below, line 171.

`data-feature` is there for controlling add tree mode, which embed disables, and hiding the add tree control if save returns a result with `enabled` set to `False`. It seems unnecessary at best, and removing it helps assure that none of these actions will be triggered.

`data-event-category` and `data-event-action` trigger google analytics, which will be triggered when the new tab or window opens in `add` mode. We probably do not want to trigger them twice.

For reasons unknown to me, "Add a Resource" doesn't have these attributes to begin with.

Exactly what change are you proposing?

Comment (jwalgran) [inline opentreemap/treemap/templates/instance_base.html]: No change needed. I was just verifying that these attributes were being hidden by design.

**Comment by RobinIsTheBird** [comment]
See also [addons #1238](https://github.com/OpenTreeMap/otm-addons/pull/1238), for treatment of suspended treemaps. This PR doesn't depend on the addons one, but the addons one depends on this.

**Comment by jwalgran** [comment]
👍

**Pull request merged**